microbootstrap/bootstrappers/base.py

```diff
@@ -85,7 +85,7 @@ def bootstrap(self) -> ApplicationT:
             dataclass_to_dict_no_defaults(self.application_config),
         )
         application = self.application_type(
-            **merge_dict_configs(resulting_application_config, self.bootstrap_before()),
+            **merge_dict_configs(self.bootstrap_before(), resulting_application_config),
```

Comment:
Do not agree with this line, cause if you want to implicitly change something on application, you can do it through `bootstrap_application`.  
This is achievable if and only if you merge application config on the last step into resulting config to rewrite some keys in config dics

Reply:
> if you want to implicitly change something on application, you can do it through bootstrap_application.

can you provide an example, please? I don't understand fully what do you mean

Reply:
to be more precise I don't understand what `bootstrap_application` is in this context

Reply:
You see, this call `self.bootstrap_before()` is returning `application_config` i.e. for litestar/fastapi.
For example, you'd like to rewrite some param that passed to config through some instrument.
In this sample we want to rewrite `param` that is passed to the application.

```python
class Settings(...):
  param = 1
  
 ...
 
 app = LitestarBootstrapper(settings).configure_application({param: 2})
```

With current realisation `{param: 2}` will be merged into config at the last moment, so it will rewrite `{param: 1}`.  
Your realisation doesn't allow such behaviour, cause we'll merge `{param: 1}` into `{param: 2}` and `configure_application` call won't do anything useful

Reply:
ok, returned merging order back. Will find another way in separate PR